### PR TITLE
Document tree sequence

### PR DESCRIPTION
**Pull request by benjeffery**
Doc treeseq methods

### CODE REVIEW

File: c/tskit/trees.h [hunk 1]
```diff
@@ -227,41 +227,418 @@ typedef struct {
...
+read from the stream --- malformed files or other errors will result in
+different error conditions. Please see the
+:ref:`sec_c_api_examples_file_streaming` section for an example of how to
+sequentially load tree sequences from a stream.
+
+Please note that this streaming behaviour is not supported if the
+TSK_LOAD_SKIP_REFERENCE_SEQUENCE option is set.
+If the TSK_LOAD_SKIP_REFERENCE_SEQUENCE option is set, the tree sequence is
+read without loading the reference sequence.
+When attempting to read from a stream with multiple tree sequence definitions
```
Comment (jeromekelleher) [inline c/tskit/trees.h:329]: Either of what two options? I'm not following the last bit here.

We should note here somewhere that the stream can be an arbitrary fd (e.g. a socket), and we don't perform any seek operations.

File: c/tskit/trees.h [hunk 2]
```diff
@@ -227,41 +227,418 @@ typedef struct {
...
+@rst
+By default the method initialises the specified destination table collection. If the
+destination is already initialised, the :c:macro:`TSK_NO_INIT` option should
+be supplied to avoid leaking memory.
+@endrst
+
+@param self A pointer to a tsk_treeseq_t object.
+@param tables A pointer to a tsk_table_collection_t object. If the TSK_NO_INIT option
+    is specified, this must be an initialised table collection. If not, it must
+    be an uninitialised provenance table.
```
Comment (jeromekelleher) [inline c/tskit/trees.h:405]: provenance table

**Review by jeromekelleher — COMMENTED** [review]
LGTM, spotted a few minor things

File: c/tskit/trees.h (hunk 1 above)
Comment (benjeffery) [reply c/tskit/trees.h:332]: Fixed

**Comment by benjeffery** [comment]
@jeromekelleher This one needs an approving review

File: c/tskit/tables.h [hunk 3]
```diff
@@ -3579,6 +3579,9 @@ If the :c:macro:`TSK_NO_INIT` option is set, the table collection is
 not initialised, allowing an already initialised table collection to
 be overwritten with the data from a file.
 
+The stream an be an arbitrary file descriptor, for example a network socket.
```
Comment (jeromekelleher) [inline c/tskit/tables.h:3582]:
```suggestion
The stream can be an arbitrary file descriptor, for example a network socket.
```

File: c/tskit/trees.h [hunk 4]
```diff
@@ -265,15 +265,188 @@ typedef struct {
 @defgroup TREESEQ_API_GROUP Tree sequence API
 @{
 */
+
+/**
+@brief Initialises the tree sequence by (optionally) copying the supplied
```
Comment (jeromekelleher) [inline c/tskit/trees.h:270]: The brief is getting to much into the options weeds I think. How about "Initialises the tree sequence based on the specified table collection."

File: c/tskit/trees.h [hunk 5]
```diff
@@ -265,15 +265,188 @@ typedef struct {
...
+flags:
+
+TSK_TAKE_TABLES
+    By default the table collection is copied, however if this flag is
+    specified the table collection will be taken and owned with a reference
+    held by this tree sequence. The table collection will be free'd with
+    this tree sequence.
+
+TSK_BUILD_INDEXES
+    If specified edge indexes will be built and stored in the table collection
```
Comment (jeromekelleher) [inline c/tskit/trees.h:290]: Is TSK_NO_CHECK_INTEGRITY an option here/

File: c/tskit/trees.h [hunk 6]
```diff
@@ -265,15 +265,188 @@ typedef struct {
...
+read without loading the reference sequence.
+
+**Options**
+
+Options can be specified by providing one or more of the following bitwise
+flags:
+
+TSK_LOAD_SKIP_REFERENCE_SEQUENCE
+    Do not load reference sequence.
+
```
Comment (jeromekelleher) [inline c/tskit/trees.h:324]: Is SKIP_TABLES an option here? What about TSK_NO_CHECK_INTEGRITY?

File: c/tskit/trees.h (hunk 5 above)
Comment (benjeffery) [reply c/tskit/trees.h:290]: No, integrity checking is always performed. I guess that could be added as an optimisation for those confident in their tables?

File: c/tskit/trees.h (hunk 4 above)
Comment (benjeffery) [reply c/tskit/trees.h:270]: Went with that and added detail in the main text.

File: c/tskit/trees.h (hunk 6 above)
Comment (benjeffery) [reply c/tskit/trees.h:328]: Neither are a valid option. We could think about adding `TSK_NO_CHECK_INTEGRITY` but `SKIP_TABLES` makes no sense I think.

File: c/tskit/trees.h (hunk 4 above)
Comment (jeromekelleher) [reply c/tskit/trees.h:270]: Doesn't look like you pushed?

Comment (benjeffery) [reply c/tskit/trees.h]: Whoops, done.

File: c/tskit/trees.h (hunk 5 above)
Comment (jeromekelleher) [reply c/tskit/trees.h:293]: That's fine, so long as the door is open for it.

File: c/tskit/trees.h (hunk 6 above)
Comment (jeromekelleher) [reply c/tskit/trees.h:327]: I think SKIP_TABLES will be accepted and work, you'll just get a tree_sequence that only has top-level stuff. This should be documented I think, as it was part of the original design that you could quickly get access to top-level stuff from either the ts or the tables interface.

Just checked, and it's tested in ``test_file_format.c``

Comment (jeromekelleher) [reply c/tskit/trees.h:327]: Maybe we could dedupe this a bit by referring to ``tsk_table_collection_load`` and say that this is basically the same, unless ``TSK_BUILD_INDEXES`` is specfied (which will additionally build the indexes)?

Same for loadf below.

**Review by jeromekelleher — COMMENTED** [review]
Still not quite right on the load docs.

Maybe we should drop an issue to track that and merge this?

File: c/tskit/trees.h (hunk 6 above)
Comment (benjeffery) [reply c/tskit/trees.h:314]: Great, didn't realise/remember that SKIP_TABLES worked in that case, having a brain foggy day.
I've added a link to the table methods.

**Comment by benjeffery** [comment]
I've added references to the table methods for `load`/`loadf`. I think we merge this and go for final cleanup/sectioning.

File: c/tskit/trees.h [hunk 7]
```diff
@@ -298,63 +436,447 @@ on and may change arbitrarily between versions.
...
+Returns an array of breakpoint locations, the array is owned by the tree sequence.
+@endrst
+
+@param self A pointer to a tsk_treeseq_t object.
+@return Returns the pointer to the breakpoint array.
+*/
 const double *tsk_treeseq_get_breakpoints(const tsk_treeseq_t *self);
+
+/**
+@brief Get the breakpoints
```
Comment (jeromekelleher) [inline c/tskit/trees.h:671]: breakpoints/samples

**Comment by jeromekelleher** [comment]
Hows this looking @benjeffery?

**Pull request merged**